test/e2e/specs/fse/fse__limited-global-styles.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @group gutenberg
+ */
+
+import {
+	DataHelper,
+	TestAccount,
+	FullSiteEditorPage,
+	SecretsManager,
+} from '@automattic/calypso-e2e';
+import { Browser, Page } from 'playwright';
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( 'Site Editor: Limited Global Styles' ), function () {
+	let page: Page;
+	let fullSiteEditorPage: FullSiteEditorPage;
+	const credentials = SecretsManager.secrets.testAccounts.simpleSiteFreePlanUser;
+	const siteSlug = credentials.testSites?.primary?.url as string;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+
+		const testAccount = new TestAccount( 'simpleSiteFreePlanUser' );
+		await testAccount.authenticate( page );
+
+		fullSiteEditorPage = new FullSiteEditorPage( page );
+	} );
+
+	it( 'Visit the site editor', async function () {
+		await fullSiteEditorPage.visit( siteSlug );
+		await fullSiteEditorPage.prepareForInteraction( { leaveWithoutSaving: true } );
+	} );
+
+	it( 'Open site styles', async function () {
+		await fullSiteEditorPage.openSiteStyles();
+	} );
+
+	it( 'Select the "Try it out" option in the upgrade modal', async function () {
+		await fullSiteEditorPage.tryGlobalStyles();
+	} );
+
+	it( 'Pick a non-default style variation', async function () {
+		await fullSiteEditorPage.setStyleVariation( 'Non-default' );
+	} );
+
+	it( 'Save the styles and check that the pre-save notice shows up', async function () {
+		// On mobile, site styles is a popover panel that blocks the Save button.
+		// So let's always close site styles first to be safe. :)
+		await fullSiteEditorPage.closeSiteStyles();
+		await fullSiteEditorPage.save( { expectedPreSaveElement: '.wpcom-global-styles-notice' } );
```

Comment:
**comment:non-blocking**: in my opinion this isn't a preferred approach for a couple of reasons:

1. it leaks the underlying interaction & operation of the framework to the test spec.
2. it is prone to failure when the selectors change.

In my opinion, a better approach would be to transparently handle the pre-save notice within the `FullSiteEdittorPage.save()` method. 

With that said, I am not going to block the merge of this PR on this comment as it is a minor issue.

Reply:
> In my opinion, a better approach would be to transparently handle the pre-save notice within the `FullSiteEdittorPage.save()` method.

The problem with that approach is that the pre-save notice doesn't appears always – it only shows up if a custom style has been set, so I don't know how `FullSiteEdittorPage.save` could handle this automatically since it doesn't know whether the spec has intentionally set a custom style or not.

For instance, the `fse__limited-global-styles.ts` spec has one step that sets a custom style (so we need to confirm that the pre-save notice shows up) and then another one that sets a default style (so we need to confirm that the pre-save notice doesn't show up).

Reply:
Alright, I came up with something to handle the pre-save notice within `FullSiteEdittorPage.save()` in https://github.com/Automattic/wp-calypso/pull/71333/commits/0a8e9c4d207cf83306177101a76de01780b10cd7

The `FullSiteEdittorPage` class knows now whether custom styles are being used or not based on the usage of the `setStyleVariation` method (so the spec no longer needs to explicitly indicate it), which allowed the `save` method to automatically determine whether the pre-save notice should be visible or not.

However, the spec still needs to pass a param to the `save` method to enable this pre-save notice check, since the `save` method is also used by other specs where we cannot deterministically determine if custom styles have been used or not (because they use other methods such as `setGlobalColorStyle` or `setGlobalLayoutStyle` ).